### ISSUE TITLE
Strange behavior with versioned relationships

### ISSUE THREAD

**Issue by faurik** [h-omer/neo4j-versioner-core#20]
Hello,

I am getting the NullPointerException when I trying to create relationship between versioned entities.

Steps to reproduce the issue:
1. Create two entities:
```
CALL graph.versioner.init('Product', {id: 123}, {name: 'Laptop', price: 299.99}, "", localdatetime('2016-04-15'));
CALL graph.versioner.init('Customer', {id: 345}, {name: 'Emma Brown', address: 'London'}, "", localdatetime('2016-12-30'));
```
2. Create a relationship between them (it will be created):
```
MATCH (customer:Customer {id: 345}), (product:Product {id: 123}) 
WITH customer, product 
CALL graph.versioner.relationship.create(customer, product,  'BUYS', {}, localdatetime('2017-01-08T11:12:35')) YIELD relationship RETURN relationship;
```
3. Create one more relationship (it will fail):
```
MATCH (customer:Customer {id: 345}), (product:Product {id: 123}) 
WITH customer, product 
CALL graph.versioner.relationship.create(customer, product,  'BUYS', {}, localdatetime('2017-03-08T11:12:35')) YIELD relationship RETURN relationship;
```
![image](https://user-images.githubusercontent.com/25117696/124080865-85640f00-da53-11eb-8ecd-7fc991993ee2.png)

Is it a bug, or do I get the principles wrong? 

---------------
Neo4j-versioner-core version 4.2.4
Neo4j server version 4.2.4

**Comment by mfalcier** [comment]
Hi @faurik, step 2 and step 3 are the same, are you making this on purpose?

**Comment by faurik** [comment]
Hi @mfalcier,

Thank you for your response.

Step 2 and step 3 are not the same. I tried to create new version of relationship (step 2 has `localdatetime('2017-01-08T11:12:35')` (January), but step 3 has `localdatetime('2017-03-08T11:12:35')` (March)). 

Probably, I misunderstand the concepts and because of that I used `neo4j-versioner-core` incorrectly, so let me rephrase my question. How can I create versioned relation between objects and how can I create new version of versioned relation?

**Comment by tpayne84** [comment]
> How can I create versioned relation between objects and how can I create new version of versioned relation?

It sounds like this is a feature request that would require a change to the current model.

Currently the `(Relationship)=FOR=>(Node)` model does not support a "versioned relation".

It does support a versioned state, `(Node)=HAS_STATE=>(StateData)`.

But to support the versioning of the relationships as well, the model would need to look like this:
`(Node)=HAS_STATE=>(StateData)<=FOR=(Relationship)`.

**Comment by faurik** [comment]
Hi @tpayne84, thank you for the clarification, at least it becomes more clear for me.
